pkg/aws/eni/node.go

```diff
@@ -452,3 +455,26 @@ func (n *Node) ResyncInterfacesAndIPs(ctx context.Context, scopedLog *logrus.Ent
 
 	return available, nil
 }
+
+// GetMaximumAllocatableIPv4 returns the maximum amount of IPv4 addresses
+// that can be allocated to the instance
+func (n *Node) GetMaximumAllocatableIPv4(scopedLog *logrus.Entry) int {
+	limits, limitsAvailable := n.getLimits()
+	if !limitsAvailable {
+		scopedLog.Warningf("Could not determined instance limits, %s", getMaximumAllocatableIPv4FailureWarningStr)
+		return 0
+	}
+
+	if limits.Adapters < *n.k8sObj.Spec.ENI.FirstInterfaceIndex {
```

Comment:
Is `FirstInterfaceIndex` defaulted when the `CiliumNode` gets created? Otherwise this will panic, right?

Reply:
Indeed, I added a check for it! 👍